### DISCUSSION TITLE
스크립트를 통해 유저의 아이디를 조회하는법

### DISCUSSION THREAD

**Discussion by JeongYire** [naverz/zepeto-studio-kor#245] [5. Q&A]
스크립트를 통해서 유저의 아이디를 조회하고싶습니다.

예를들어 장소에 특정한 유저가 들어왔을경우

그 유저의 아이디를 조회해서 화면에 띄운다던지요. 

제페토스튜디오 내에서 지원하는 ZepetoWorldHelper.GetUserInfo 는 
이미 UserID 를 알고있어야 가능한 함수인지라

이 경우엔 어떻게 유저의 아이디를 조회할 수 있을지 문의드려봅니다

일단은 OnTriggerEnter 를 통해 유저의 아이디를 얻는걸 목표로 하고있습니다

**Comment by hyeonjin-jo** [comment]
안녕하세요.

여러 가지 방법이 있을 수 있지만 본문에 적힌 내용만 보고 조언을 드려보자면,

특정 장소에 유저가 들어왔을 경우(충돌 체크 또는 위치값) 서버로 해당 유저의 아이디를 send하고

클라이언트에서 receive 하는 방식이면 가능할 것 같습니다.

**Reply by JeongYire** [reply]
그걸 모르겠습니다. 충돌 함으로써 어떻게 해당 유저의 아이디를 갖고오죠? OnTriggerEnter 로 GameObject 는 갖고올수있어도
그 게임오브젝트가 어떤 아이디를 가진 유저의 오브젝트인지 알 수 있는 방법이 필요합니다.

**Reply by hyeonjin-jo** [reply]
유저아이디의 경우 서버쪽에서 OnJoin에서 획득이 가능합니다.

UserId값을 캐싱하고 있다가 OnTriggerEnter에서 활용하면 좋을 것 같습니다.

**Reply by JeongYire** [reply]
OnJoin 단에서 미리만들어놓은 오브젝트에  유저정보를 담아놓고 캐릭터 오브젝트의 자식으로 넣었더니 성공하였습니다 감사합니다
